frontend/benefit/handler/src/utils/calculator.ts

```diff
@@ -32,3 +32,44 @@ export const extractCalculatorRows = (
     helsinkiBenefitMonthlyRows,
   };
 };
+
+// eslint-disable-next-line unicorn/prefer-set-has
+const SUBTOTAL_CALCULATION_ROW_TYPES = [
+  CALCULATION_ROW_TYPES.HELSINKI_BENEFIT_MONTHLY_EUR,
+  CALCULATION_ROW_TYPES.HELSINKI_BENEFIT_SUB_TOTAL_EUR
+];
+
+export const groupCalculatorRows = (rows: Row[]): Row[][] => {
```

Comment:
It's not an elegant design, that's for sure. By design, this should however only isolate the two very specific cases into their own dedicated sections (consecutive subtotal rows and date range headings) and leave everything else into the fallback ones. Only the subtotal sections get any sort of special treatment afterwards either way.

Do you suppose this may not work for even that task in all cases? In that case, any alternative ideas? If we lack good alternative options, implementing the design from Abstract as is may prove untenable.

Reply:
FWIW, from one of my already existing test alterations, which already had an apprenticeship and a subsidy:

![image](https://github.com/City-of-Helsinki/yjdh/assets/12729676/cabfb1d0-f7c7-4deb-bd6b-0eb2a2cba990)